Serverless pattern for generating images using Amazon Bedrock and saving to Amazon S3

Chrispyontheoutside opened the pull request:
*Issue #, if available:*

*Description of changes:*
Serverless pattern to deploy an API that generates an image using Amazon Bedrock (Stable Diffusion XL).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

ellisms commented:
Hi @Chrispyontheoutside - the `example-pattern.json` should be in the root directory of the project, but I see it in `apigw_lambda_bedrock/`, can you move it to `apigw-custom-domain-regional` and submit a revision?

Chrispyontheoutside commented:
Hi @ellisms, thanks for the review! I moved it to the root directory of the project.

ellisms commented:
@Chrispyontheoutside I tested the pattern today and have the following suggestions/questions:

- In the python venv activation, I would also include instructions for those on Mac/Linux.
- The pattern and instructions mention an S3 bucket is created where you can view the output. When I ran my cdk deploy, it did not create a bucket and the Lambda function doesn't appear to attempt to save output to a bucket. Instead the output is returned via the API call. I do see the bucket definition in cdk, but it doesn't seem to be created. Did you intend to not use the bucket? 
- If you do decide to keep the bucket, I would add it's name to the outputs so the user can find it easier.

Chrispyontheoutside commented:
Hi @ellisms thanks for testing, I've made the following changes:
- Updated the README for installing the venv on MacOS/Linux
- The bucket name is outputted in the CDK output
- The function saves the image to the root directory of the bucket instead of /tmp/.

Thanks!

ellisms commented:
@Chrispyontheoutside I figured out why I was having so much trouble testing (and not seeing code updates), after cloning the repo, the change directory step has the wrong path, it should be `apigw-lambda-bedrock-s3-cdk-python`. When I use this directory, I have the following feedback:
* Update the directory name in the instructions
* In order to use Python3.11 in the template, the aws-cdk-lib version needs to be at least 2.88.0. Please update the requirements file.
* The instructions say you need access to Anthropic/Claude, but the code calls StabilityAI. 
* I'm getting an error when running the curl command against APIGW: `{"message":"Missing Authentication Token"}`. If I test the endpoint in the APIGW console, it works just fine and produces the image.

Chrispyontheoutside commented:
Hi @ellisms, thanks for the review, here are my following changes for the feedback:

- updated the directory to apigw-lambda-bedrock-cdk-python-s3
- updated the requirements file
- changed to Stability AI / model id for SDXL 1.0
- updated the curl command for the resource

Please let me know if you have any other feedback, thanks!

ellisms commented:
Thanks, I tested and everything works. Take a look at the comments I left, some are already taken care of, but a few still need to be addressed. After that, I think we'll be set.

Chrispyontheoutside commented:
Hi, @ellisms. Thank you! I'm not seeing the conversation comments. Apologies if i'm looking in the wrong area.

ellisms commented:
Sorry, I never finalized the review. Do you see it now?

Chrispyontheoutside commented:
Yup! Thanks I'll clear these out.

Chrispyontheoutside commented on apigw-lambda-bedrock-s3-cdk-python/README.md:
```diff
@@ -0,0 +1,125 @@
+
+## Review results
+
+Go to Amazon S3, and navigate to the S3 bucket. It will have a name similar to 'apigwlambdabedrockstack-mybucket...'. In the directory `\tmp\` you should see `image-{num}.png`. 
```
Returning file_name in lambda
